### PR TITLE
DDF-4668 Fix Server Side Paging Indices

### PR DESCRIPTION
**Pull request by Bdthomson**
#### What does this PR do?
- Fixes how server side page indexes are calculated and stored.
- Fixes CSS for the "Update with new data?" button to prevent the text from getting cut off.

#### Who is reviewing it? 

@adimka 
@bdeining 

#### Select relevant component teams: 
@codice/ui 

#### Ask 2 committers to review/merge the PR and tag them here.

@andrewkfiedler 
@djblue

#### How should this be tested?
1. Setup multiple federated sources and set a page size of 5, 10, and 25 to ensure that you get the correct results.
2. Setup multiple federated sources where one returns results slower, and try the same page sizes, making sure that you wait until the slower source returns and you have the option to click the "Update with new data?" button.

#### What are the relevant tickets?

For GH Issues:
Fixes: #4668

#### Screenshots
image

#### Checklist:
- [ ] Documentation Updated
- [ ] Update / Add Threat Dragon models
- [ ] Update / Add Unit Tests
- [ ] Update / Add Integration Tests

#### Notes on Review Process
Please see [Notes on Review Process](https://codice.atlassian.net/wiki/spaces/DDF/pages/71946981/Pull+Request+Guidelines) for further guidance on requirements for merging and abbreviated reviews. 

#### Review Comment Legend:
- ✏️ (Pencil) This comment is a nitpick or style suggestion, no action required for approval. This comment should provide a suggestion either as an in line code snippet or a gist. 
- ❓ (Question Mark) This comment is to gain a clearer understanding of design or code choices, clarification is required but action may not be necessary for approval.
- ❗ (Exclamation Mark) This comment is critical and requires clarification or action before approval.

### CODE REVIEW

**Comment by bdeining** [comment]
build now

File: ui/packages/catalog-ui-search/src/main/webapp/js/model/Query.js [hunk 1]
```diff
@@ -75,6 +75,53 @@ const handleTieredSearchLocalFinish = function(ids) {
...
+  }
+}
+
+const currentIndexForSource = state => {
+  return state.reduce((counts, page) => {
+    return Object.keys(page).reduce((counts, src) => {
+      const resultCount = page[src]
+
+      if (counts[src] === undefined) {
+        counts[src] = 1
```
Comment (samuelechu) [inline ui/packages/catalog-ui-search/src/main/webapp/js/model/Query.js]: counts[src] = 0 ? could you use a named function here to make things more readable?

File: ui/packages/catalog-ui-search/src/main/webapp/js/model/Query.js [hunk 2]
```diff
@@ -157,15 +207,31 @@ Query.Model = PartialAssociatedModel.extend({
...
-    )
     this.listenTo(this, 'change:cql', () => this.set('isOutdated', true))
+
+    const sync = () => {
+      this.dispatch(updateResults(this.get('result').toJSON()))
+      this.set('serverPageIndex', serverPageIndex(this.state))
+
+      const totalHits = this.get('result')
+        .get('status')
+        .reduce((total, status) => {
```
Comment (samuelechu) [inline ui/packages/catalog-ui-search/src/main/webapp/js/model/Query.js]: ✏️ can you do `total = 0` instead of the initial 0 value?

Comment (Bdthomson) [reply ui/packages/catalog-ui-search/src/main/webapp/js/model/Query.js]: That's generally not how reduce functions are written.

File: ui/packages/catalog-ui-search/src/main/webapp/js/model/Query.js (hunk 1 above)
Comment (Bdthomson) [reply ui/packages/catalog-ui-search/src/main/webapp/js/model/Query.js]: We are starting the count at 1 if the src is found.

**Comment by willwill96** [comment]
### Hero Successful
Setup multiple federated sources, some slower than others. Verified correct results with varying of paging setups

LGTM

**Pull request merged**